common/trinary/trit_long.c

```diff
@@ -39,60 +46,72 @@ size_t long_to_trits(int64_t value, trit_t *out) {
     if (negative) {
       trit = -trit;
     }
-
-    out[i] = trit;
+    trits[i] = trit;
     value++;
     value /= RADIX;
   }
   return size;
 }
 
-size_t nearest_greater_power_of_three(size_t value, size_t *power) {
-  size_t vp = 1;
-  *power = 0;
-  while (value > vp) {
-    *power += 1;
-    vp *= RADIX;
-  }
-  return vp;
-}
-
-size_t encoded_length_variables(int64_t value, size_t *start, size_t *power) {
-  size_t size, size_size;
-  size = nearest_greater_power_of_three(min_trits(value), power);
-  size_size = (*power + 1) / 2;
-  *start = size_size;
-  return size_size + size;
+size_t nearest_greater_multiple_of_three(size_t value) {
+  size_t rem = value % RADIX;
+  if (rem == 0) return value;
+  return value + RADIX - rem;
 }
 
 size_t encoded_length(int64_t value) {
-  size_t start, power;
-  return encoded_length_variables(value, &start, &power);
+  if (value == 0) return sizeof(encoded_zero) / sizeof(trit_t);
+  size_t length = nearest_greater_multiple_of_three(min_trits(llabs(value)));
+  // trits length + encoding length
+  return length + min_trits(pow(2, length / RADIX) - 1);
```

Comment:
A left shift by n bits is defined as multiplication with pow(2,n). Rewriting with `1 << (length / RADIX)` would be slightly concise.